### ISSUE TITLE
Allow config file entries to be read from environment

### ISSUE THREAD

**Issue by yeastplume** [OliveTin/OliveTin#350]
I'm trying to configure a small network of services via docker/kubernetes and use olivetin to monitor them. I have a few different configurations for test and deployment, and it would be very useful to have the ability to replace config file entries with environment variables. For instance, I'd like to use something like:

`listenAddressSingleHTTPFrontend: ${NODE_MONITOR_LISTEN_ADDRESS}`

Is there a way to do this already? The only alternative seems to be keeping multiple copies of config files with mostly redundant info, or doing a manual merge step somewhere which doesn't sit very nicely in the docker workflow.

**Comment by jamesread** [comment]
Hey, OliveTin doesn't read any configuration instructions from environment variables, and it only has one flag - for confdir. 

Ome day I'd like to allow multiple config files to be read from a directory, but that isn't possible today. 

I am open to adding configuration options via environment variables, please list the environment variables you think you'll need and we'll see if the design is easy enough.

**Comment by yeastplume** [comment]
Thanks for responding! I think most cases can be covered via commands calling shell scripts that read environment variables, but it's generally not possible to configure anything related to ports or bind addresses, which are generally what might need to change between different deployment environments. So really, if we had the ability to specify anything port related via environment var I think that would cover a lot of deployment use cases here.

So basically, all of these (unless there are some more that I haven't found)

https://docs.olivetin.app/network-ports.html

**Comment by jamesread** [comment]
Okay dokey, are you installing several OliveTin insurances on one server? Because this shouldn't be necessary for docker containers right?

**Comment by yeastplume** [comment]
Agreed, but it does come up when I'm running development containers with a few environments, and on this project I'm trying to keep ports organised in a certain way. Not critical, but most services I've come across allow bind addresses to be configured from outside the container environment, usually though env vars passed to command line switches. Reading from configuration files directly like I originally suggested would allow for other subtle differences to exist between test/production environments from the same config file, but I think just allowing for bind address/port configuration via command line would be good first addition.

**Comment by jamesread** [comment]
@yeastplume This has now been implemented in the 2024.07.15 version. Could you upgrade, try setting the PORT environment variable and let me know if this works for you as expected.

**Comment by yeastplume** [comment]
> @yeastplume This has now been implemented in the 2024.07.15 version. Could you upgrade, try setting the PORT environment variable and let me know if this works for you as expected.

Yes, that looks good and I'm able to specify the port in a docker-compose file via the environment. Thanks for the change!

**Comment by mlhynfield** [comment]
Hi! This may warrant a separate issue, but I'm new to this lovely project and want to test this out in my Kubernetes-based home lab. I already have Authentik running as SSO for most of my self-hosted services, so I was quite excited to see [Authentik OAuth2 support](https://docs.olivetin.app/oauth2-authentik.html). I deploy and maintain my services using [Argo CD](https://argo-cd.readthedocs.io/en/stable/) for GitOps, so I began setting up OliveTin with an app-template Helm chart (the chart associated with this project, while good, is a little lacking in desired configuration options currently; that's neither here nor there), until I realized I didn't know how to pass the Authentik client secret to OliveTin without checking it in Git in plaintext as part of the ConfigMap:

> authOAuth2Providers:
  authentik:
    name: authentik
    title: Authentik
    clientID: "1234567890"
    clientSecret: "123456789012345"

As I said, I'm new to this project, so I couldn't say for sure, but theoretically I could imagine there might be other scenarios where secret/sensitive information might need to be passed through the OliveTin config. Regardless, it would be great to at least have the option to pass this OAuth2 client secret in a different manner, be that by environment variable or something else.

Thanks for your consideration, and keep up the great work! :)

**Comment by mlhynfield** [comment]
Looking into the docs a little further, I think something similar to the existing entity var string replacement behavior would also work great:

https://github.com/OliveTin/OliveTin/blob/8d4ddd36cda7d2667a651ff326e195878c5840fa/internal/stringvariables/entities.go#L13-L31

The ability to reference variable values from other files in the `config.yaml` should work nicely with Secret file injection in Kubernetes, and probably also with docker-compose secrets.
